source/extensions/filters/http/jwt_authn/filter.h

```diff
@@ -44,9 +43,7 @@ class Filter : public Http::StreamDecoderFilter,
   State state_ = Init;
   // Mark if request has been stopped.
   bool stopped_ = false;
-  std::vector<AsyncMatcherSharedPtr> rule_matchers_;
-  Thread::MutexBasicLockable lock_;
-  std::size_t count_ GUARDED_BY(lock_);
+  std::vector<MatcherConstSharedPtr> rule_matchers_;
```

Comment:
Can this matcher vector be created and stored in FilterConfig?  and just pass FilterConfig to each Filter.